### ISSUE TITLE
_FakeObjectResolver cannot deal with `null`

### ISSUE THREAD

**Issue by sma** [NicolaVerbeeck/dart_pdf_reader#33]
I'm trying to parse my collection of  TTRPG PDFs to test the parser and got an exception in
```dart
class _FakeObjectResolver implements ObjectResolver {
  ...
  @override
  Future<T?> resolve<T extends PDFObject>(PDFObject? toResolve) =>
      Future.value(toResolve as T);
}
```
because it tries to cast a `null` value to `T` (which is a `PDFObject` in this case) while trying to resolve

```
PDFDictionary (<</Type /XRef /Root 1 0 R /Info 2 0 R /ID [<3dfecacc0c60d134cc717df19d0964ec> <3dfecacc0c60d134cc717df19d0964ec>] /Size 419 /W [1 4 2] /Filter /FlateDecode /Length 1106>>)
```

because there are no `1 0 obj` or `2 0 obj` defined in this file. The first obj is `4 0`. 

You could get the PDF for free [from here](https://www.drivethrurpg.com/product/258103/Liminal-Quickstart) but it's probably enough to know that there are somewhat corrupt PDFs out there :(

**Comment by NicolaVerbeeck** [comment]
Good find, subtle bug. Fix incoming

**Comment by NicolaVerbeeck** [comment]
Fix in 0.4.0
